third_party/intel/lib/TritonIntelGPUToLLVM/Utility.h

```diff
@@ -747,19 +754,32 @@ loadSharedToDistributed(Value dst, Value src, SharedMemoryObject &shrMemObj,
   auto srcTy = cast<MemDescType>(src.getType());
 
   SmallVector<Value> ret;
-  if (emitTransferBetweenRegistersAndShared(
-          dstTy, srcTy, elemTy, /*maxVecElems=*/std::nullopt,
-          shrMemObj.getBase(), shrMemObj.getStrides(), loc, rewriter, target,
-          [&](VectorType vecTy, Value vecAddr) {
-            auto vecVal = load(vecTy, vecAddr);
-            vecVal.setAlignment(vecTy.getNumElements() *
-                                elemTy.getIntOrFloatBitWidth() / 8);
-
-            for (int v = 0; v < vecTy.getNumElements(); v++) {
-              ret.push_back(extract_element(elemTy, vecVal, i32_val(v)));
-            }
-          }))
+  if (auto dpasLayout = dyn_cast<DpasEncodingAttr>(dstTy.getEncoding())) {
+    if (emitTransferBetweenDPASAndShared(
+            dstTy, srcTy, elemTy, /*maxVecElems=*/std::nullopt,
+            shrMemObj.getBase(), shrMemObj.getStrides(), loc, rewriter, target,
+            [&](VectorType vecTy, Value vecAddr) {
+              auto vecVal = load(vecTy, vecAddr);
+              vecVal.setAlignment(vecTy.getNumElements() *
+                                  elemTy.getIntOrFloatBitWidth() / 8);
+              for (int v = 0; v < vecTy.getNumElements(); v++) {
+                ret.push_back(extract_element(elemTy, vecVal, i32_val(v)));
+              }
+            }))
+      return ret;
+  } else if (emitTransferBetweenRegistersAndShared(
+                 dstTy, srcTy, elemTy, /*maxVecElems=*/std::nullopt,
+                 shrMemObj.getBase(), shrMemObj.getStrides(), loc, rewriter,
+                 target, [&](VectorType vecTy, Value vecAddr) {
+                   auto vecVal = load(vecTy, vecAddr);
+                   vecVal.setAlignment(vecTy.getNumElements() *
+                                       elemTy.getIntOrFloatBitWidth() / 8);
+                   for (int v = 0; v < vecTy.getNumElements(); v++) {
+                     ret.push_back(extract_element(elemTy, vecVal, i32_val(v)));
+                   }
+                 })) {
     return ret;
+  }
```

Comment:
```suggestion
  }
  bool success = emitTransferBetweenRegistersAndShared(
                 dstTy, srcTy, elemTy, /*maxVecElems=*/std::nullopt,
                 shrMemObj.getBase(), shrMemObj.getStrides(), loc, rewriter,
                 target, [&](VectorType vecTy, Value vecAddr) {
                   auto vecVal = load(vecTy, vecAddr);
                   vecVal.setAlignment(vecTy.getNumElements() *
                                       elemTy.getIntOrFloatBitWidth() / 8);
                   for (int v = 0; v < vecTy.getNumElements(); v++) {
                     ret.push_back(extract_element(elemTy, vecVal, i32_val(v)));
                   }
                 });
  if (!success)
    llvm::report_fatal_error("Failed to emit transfer from shared to register");
```
and then remove the code after.